pkg/compactor/job_test.go

```diff
@@ -90,6 +96,15 @@ func TestJobWaitPeriodElapsed(t *testing.T) {
 			expectedElapsed: true,
 			expectedMeta:    nil,
 		},
+		"out of order block": {
+			waitPeriod: 10 * time.Minute,
+			jobBlocks: []jobBlock{
+				{meta: meta5, attrs: objstore.ObjectAttributes{LastModified: time.Now().Add(-20 * time.Minute)}},
+				{meta: meta6, attrs: objstore.ObjectAttributes{LastModified: time.Now().Add(-15 * time.Minute)}},
```

Comment:
Wouldn't this test pass anyway given the wait period is 10m but both meta5 and meta6 have been uploaded since more than 10m?

```suggestion
				{meta: meta6, attrs: objstore.ObjectAttributes{LastModified: time.Now().Add(-5 * time.Minute)}},
```

Reply:
Well spotted